lib/Parse/ParseVersion.cpp

```diff
@@ -21,10 +21,10 @@ swift::version::Version version::getCurrentCompilerVersion() {
 #ifdef SWIFT_COMPILER_VERSION
   auto currentVersion = VersionParser::parseVersionString(
       SWIFT_COMPILER_VERSION, SourceLoc(), nullptr);
-  assert(currentVersion.hasValue() &&
+  assert(static_cast<bool>(currentVersion) &&
```

Comment:
Any reason not to just use `has_value` and `get_value`?

Reply:
That would work too